tests/tests/swfs/avm2/prototype_set_null/Test.as

```diff
@@ -0,0 +1,14 @@
+package
+{
+    import flash.display.Sprite;
+
+    public class Test extends Sprite
+    {
+    }
+}
+
+var val:Function = new Function();
+
+val.prototype = null;
```

Comment:
cool, I had no idea this was possible :)
Could you expand the test a bit more, to make sure that instantiation works and the resulting object is usable? Like
```as
function f(){}
f.prototype = null;
var x = new f();
trace(x); // [object Object]
trace(x.asdf); // undefined
```

Reply:
Can do.

Reply:
```
2023-07-24T14:21:57.091519Z  WARN run_frame: ruffle_core::display_object::movie_clip: Got AVM2 error TypeError: Cannot convert <Unknown type> to primitive.
	at /trace()
	at global$init() [TU=Test] when attempting to assign symbol class Test
```
Interestingly it fails in Ruffle at the first `trace()` call, but works in Flash Player.

Reply:
```
function f():void {};

f.prototype = null;

var x = new f();

trace(x.prototype); // undefined

trace(x.valueOf()); // [object Object]
trace(x.toString()); // [object Object]

trace(x); // [object Object]
trace(x.asdf); // undefined
```
It seems when `prototype` is set to undefined in Flash Player `valueOf` and `toString` methods remain, but in Ruffle they disappear?

Ruffle Output:
```
2023-07-24T14:45:23.087507Z  WARN run_frame: ruffle_core::display_object::movie_clip: Got AVM2 error TypeError: Error #1006: valueOf is not a function of class <Unknown type>.
	at global$init() [TU=Test] when attempting to assign symbol class Test
```